composer/trainer/checkpoint.py

```diff
@@ -393,17 +405,27 @@ class CheckpointSaver:
     """Manager for saving trainer state to checkpoint files.
 
     Args:
-        save_folder (str): The path to store checkpoints in.
+        save_folder (str): The folder to store checkpoints in. The ``save_folder`` will be relative
+            to the folder returned by :meth:`~composer.utils.run_directory.get_run_directory`.
+            If the ``save_folder`` does not exist, it will be created.
         interval (Time or str): The amount of time units to wait between checkpoints.
         compression (str, optional): Compression algorithm to run on checkpoints. Can be ``gzip``, ``bzip2``,
             ``lzma``, or ``None`` for no compression.  (default: ``None`` for no compression).
 
     Attributes:
+        checkpoint_folder (str): The folder in which checkpoints are stored. This folder is relative
+            to the run directory of the training run. If no run directory is proivded, then by default,
+            it is of the form ``runs/<timestamp>/rank_<GLOBAL_RANK>/<save_folder>`` where ``timestamp``
+            is the the start time of the run in iso-format, ``GLOBAL_RANK`` is the global rank of the process,
+            and ``save_folder`` is the save_folder argument provided upon construction.
+
+            .. seealso:: :mod:`~composer.utils.run_directory` for details on the format of the run directory
+                and how to customize it.
         saved_checkpoints (Dict[Timestamp, List[str]): A dictionary mapping a save timestamp
             to a list of filepaths corresponding to the checkpoints saved at that time.
 
             .. note:: The list of filepaths is for all ranks. The path at index ``i`` is the
```

Comment:
```suggestion
            .. note:: When using DeepSpeed, the index of a filepath in each list corresponds to the global rank of the process that wrote that file. These filepaths are valid only on the global rank's node. Otherwise, when not using DeepSpeed, this list will contain only one filepath since only rank zero saves checkpoints.
```